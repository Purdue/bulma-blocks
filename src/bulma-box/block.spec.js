jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    getAllBlocks,
    selectBlockByClientId,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    clickElementByText,
    blockStartup,
    clickCheckbox,
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Box', blockName: 'bulma-blocks/box'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    test( 'ðŸ”Ž Add inner block button.', async () => {
        await blockStartup(block)

        const innerBlockButton = await page.$('.block-editor-button-block-appender[aria-label="Add block"]')
        await innerBlockButton.click()

        const blockButton = await page.$('.components-button.editor-block-list-item-paragraph')
        await blockButton.click()

        expect(await page.$(`[data-type="core/paragraph"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    })
})