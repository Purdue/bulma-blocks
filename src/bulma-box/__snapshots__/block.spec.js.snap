// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ”¬ Bulma - Box Block ðŸ”Ž Block should be available. 1`] = `
"<!-- wp:bulma-blocks/box -->
<div class=\\"box\\"></div>
<!-- /wp:bulma-blocks/box -->"
`;
